bench/cardano-topology/src/Cardano/Benchmarking/Topology/Types.hs

```diff
@@ -0,0 +1,95 @@
+{-# LANGUAGE DeriveGeneric #-}
+
+--------------------------------------------------------------------------------
+
+module Cardano.Benchmarking.Topology.Types (
+    Topology (..)
+  , Node (..)
+  , Location (..)
+  , AWSRegion (..)
+) where
+
+import           Prelude
+
+import           GHC.Generics
+
+import qualified Data.Text as Text
+import qualified Data.Aeson as Aeson
+
+--------------------------------------------------------------------------------
+
+-- | A topology as it's used to define benchmarking profiles.
+data Topology = Topology
+  { coreNodes  :: [Node]
+  , relayNodes :: [Node]
+  }
+  deriving (Eq, Show, Generic)
+
+instance Aeson.ToJSON Topology
+
+instance Aeson.FromJSON Topology
+
+--------------------------------------------------------------------------------
+
+-- | A node as it's used to define benchmarking profiles.
+--   These nodes don't have remote addresses allocated, are referenced by name.
+data Node = Node
+  { name      :: String
+  , nodeId    :: Int -- TODO: Swith to Word64 like Cardano.Topology ?
+  -- ^ Placement, not present in Cardano.Node.Configuration.Topology.
+  , region    :: Location
+  -- ^ As Cardano.Node.Configuration.Topology but names not addresses.
+  , producers :: [String]
+  -- ^ Actually always "IOHK".
+  , org       :: String
+  , pools     :: Maybe Int
+  , stakePool :: Maybe Bool
+  }
+  deriving (Eq, Show, Generic)
+
+instance Aeson.ToJSON Node where
+{--
+  toEncoding = Aeson.genericToEncoding
+    Aeson.defaultOptions
+    { Aeson.omitNothingFields = True }
+--}
+
+instance Aeson.FromJSON Node
+
+--------------------------------------------------------------------------------
+
+-- | Location is either "loopback" used for local runs or an AWS Region name
+-- used for cloud runs.
+data Location = Loopback | AWS AWSRegion
+  deriving (Eq, Show, Generic)
+
+instance Aeson.ToJSON Location where
+  toJSON Loopback = Aeson.toJSON $ Text.pack "loopback"
+  toJSON (AWS r)  = Aeson.toJSON r
+
+instance Aeson.FromJSON Location where
+  parseJSON v =
+    (Aeson.withText "Location" $ \t -> case Text.unpack t of
+      -- Either "loopback" or an already defined AWS Region.
+      "loopback" -> return Loopback
+      _ -> AWS <$> Aeson.parseJSON v
```

Comment:
Is there a reason not to use `instance Eq Text`, i.e. `if t == "loopback" then ... else ...`? You woldn't need to unpack.

Reply:
With the previous OverloadedStrings comment I can also resolve this one